crates/bevy_ecs/src/schedule/ambiguity_detection.rs

```diff
@@ -0,0 +1,640 @@
+            relations.union_with(&dependencies);
+            relations.union_with(&dependants);
+            relations.insert(index);
+            relations
+        })
+        .collect::<Vec<FixedBitSet>>();
+    let mut ambiguities = Vec::new();
+    let full_bitset: FixedBitSet = (0..systems.len()).collect();
+    let mut processed = FixedBitSet::with_capacity(systems.len());
+    for (index_a, relations) in all_relations.drain(..).enumerate() {
+        // TODO: prove that `.take(index_a)` would be correct here, and uncomment it if so.
+        for index_b in full_bitset.difference(&relations)
+        // .take(index_a)
+        {
+            if !processed.contains(index_b)
+                && !should_ignore_ambiguity(systems, index_a, index_b, crates_filter, report_level)
+            {
+                let a_access = systems[index_a].component_access();
+                let b_access = systems[index_b].component_access();
+                if let (Some(a), Some(b)) = (a_access, b_access) {
+                    let conflicts = a.get_conflicts(b);
+                    if !conflicts.is_empty() {
+                        ambiguities.push(SystemOrderAmbiguity {
+                            system_a_index: index_a,
+                            system_b_index: index_b,
+                            conflicts,
+                        });
+                    }
+                } else {
+                    ambiguities.push(SystemOrderAmbiguity {
+                        system_a_index: index_a,
+                        system_b_index: index_b,
+                        conflicts: Vec::default(),
+                    });
+                }
+            }
+        }
+        processed.insert(index_a);
+    }
+    ambiguities
+}
+
+impl Default for ReportExecutionOrderAmbiguities {
+    fn default() -> Self {
+        ReportExecutionOrderAmbiguities::Minimal
+    }
+}
+
+impl SystemStage {
+    /// Returns all execution order ambiguities between systems
+    ///
+    /// Returns 4 vectors of ambiguities for each stage, in the following order:
+    /// - parallel
+    /// - exclusive at start,
+    /// - exclusive before commands
+    /// - exclusive at end
+    ///
+    /// # Panics
+    ///
+    /// You must call [`SystemStage::initialize`] first or this method will panic.
+    fn ambiguities(
+        &self,
+        report_level: ReportExecutionOrderAmbiguities,
+    ) -> [Vec<SystemOrderAmbiguity>; 4] {
+        if report_level == ReportExecutionOrderAmbiguities::Off {
+            return [
+                Vec::default(),
+                Vec::default(),
+                Vec::default(),
+                Vec::default(),
+            ];
+        }
+
+        // System order must be fresh
+        debug_assert!(!self.systems_modified);
+
+        // TODO: remove all internal ambiguities and remove this logic
+        let ignored_crates = if report_level != ReportExecutionOrderAmbiguities::ReportInternal {
```

Comment:
The alternative strategy, of making this a configurable field on a per plugin basis or otherwise enabling this resulted in a much more complex, less ergonomic API.

The hard-coded list still existed, it was just scattered across a dozen different files.